### PR TITLE
Q&A 3: Reader questions, corrections, and source-labeled candidate bank

### PR DESCRIPTION
**Pull request by natolambert**
## Summary

Drafts the **Q&A 3** deck (`teach/course/qa-03.md`) with four answered reader questions and one correction:

1. **"RL is basically just regularized SFT" papers** (Lecture 3) — the shared gradient algebra is real, but it leaves out negative advantages, on-policy data, and the small-scale evidence regime.
2. **Should capabilities be trained separately, then merged?** (Lecture 10) — increasingly yes as an engineering decomposition: train specialist teachers, then consolidate behavior with MOPD rather than serial finetuning or parameter averaging. Includes the teacher-support caveat from Nemotron 3 Ultra.
3. **Why does RLVR often omit the KL penalty?** (Lecture 10) — exact verifiers reduce the learned-reward proxy problem, while fixed-reference KL penalizes useful novelty and costs another model forward pass. The answer also covers incomplete verifiers, capability retention, regressions, and behavior-policy controls.
4. **Prompt-budget pocket rules** (Lecture 2) — the ~1M SFT prompt figure holds from ~4B up; larger models need a bit less as easy data stops helping, and the mixture moves more than the count.
5. **Correction: Lecture 1 rejection-sampling argmax** — `argmax_j` returns an index, not a completion. The source fix already merged in #481.

## Candidate question bank

Adds seven deliberately removable, answer-TBD candidate slides. Every slide includes a visible source channel, author/thread link, and evidence status so the bank is easy to filter later:

- **YouTube:** strict-DSL collapse; when in-house traces belong in weights rather than ICL; distillation-only continued pretraining.
- **GitHub:** reward-model sweep objectives (#337); chat templates for preference RMs (#450); transfer of high-entropy "fork token" updates (#398); and the pending `k3` GRPO instability rerun (#489).

The GitHub-derived prompts are explicitly caveated where the source is an implementation discussion, a limited reproduction, or an unresolved single-seed experiment.

Also adds the Q&A 3 card to the course page (PDF/Slides/Source, no Watch link until the video is recorded).

## Status

- Deck builds clean with `make course-lecture-qa-03`
- 29 slides; every exported PDF page visually checked for clipping, overflow, and legibility
- Local HTML preview and an embedded image both return HTTP 200
- All bibliography citations resolve against `refs.bib`
- Draft until the deck is reviewed/recorded — then add the YouTube Watch link to the card

### CODE REVIEW

**Comment by natolambert** [comment]
## Public-engagement scan: candidate questions for Q&A 3

> **Decision:** Added the first two questions in commit `b6c4104`. The evaluator-population and tool-output candidates were not selected; they were too broad or underspecified for this deck.

I scanned 25 recent YouTube uploads (including all 17 current course videos) and 254 public comments, then checked public LinkedIn posts/comments, Hacker News/Reddit, Interconnects, Bluesky/X mirrors, and recent GitHub issues/PR discussions. I deduped against Q&A 1, Q&A 2, and this PR's current three topics.

### My cut: best additions to this deck

1. **Should capability post-training be staged by mode, then merged later?**  
   If coding, math, and tool use are distinct modes, when does stagewise / multi-teacher on-policy distillation beat one joint mixture? Nathan gave the punchline ("a bunch of labs do something similar now") but the interference, forgetting, evidence, and merge mechanics remain unanswered. [Lecture 10 thread](https://www.youtube.com/watch?v=IwpYxANrpUs&lc=UgwCZ6w8z4aPayBK3pB4AaABAg)

2. **Why does RLVR often omit the KL penalty—and when should it come back?**  
   Is verifiable ground truth enough to avoid learned-reward proxy overoptimization? The thread only got a directional "Yeah is about right"; verifier misspecification, drift, collapse, and long-horizon/agentic cases would make the real answer. [Lecture 10 follow-up](https://www.youtube.com/watch?v=IwpYxANrpUs&lc=UgwCZ6w8z4aPayBK3pB4AaABAg.AZq-x8qDJuuAZq0_-kE-33)

3. **Can a bounded evaluator population represent a massively heterogeneous user base?**  
   As deployment scales, when does representativeness stop being a labeling problem and become an architectural/product problem? This is a strong bridge from preference-data mechanics to what the learned objective actually means. [Kenneth Reidbord on LinkedIn](https://www.linkedin.com/posts/natolambert_my-book-reinforcement-learning-from-human-activity-7448063772436484096-9C5h)

4. **Should tool outputs be masked from the training loss?**  
   What evidence compares masking environment/tool tokens with allowing the loss to train on them, and what changes for multi-turn credit assignment? Crisp, recent, and unanswered. [Lecture 11 comment](https://www.youtube.com/watch?v=GMry2DzC304&lc=UgwYoV-Kk8HsRpxduJ14AaABAg)

The first two questions fit the deck's RL/SFT/regularization arc and are now included. The evaluator-population and tool-output questions were left out as weaker fits.

### Strong next tier

- **How do you debug collapse with sparse rewards and a strict DSL?** Why can negative rewards for invalid syntax fail—or accelerate collapse? [Lecture 4](https://www.youtube.com/watch?v=i-AIMpZHgeg&lc=UgyHn_MoFfktNiZK4cJ4AaABAg)
- **When should proprietary traces be trained into weights rather than supplied through ICL?** Is the in-house-RL case quality, example-selection limits, cost/latency, or all three? [Lecture 7](https://www.youtube.com/watch?v=6nyJ8y8ghsE&lc=Ugy8yP4TjQX4_cHJx3N4AaABAg)
- **What metric should select reward-model hyperparameters?** Do rankings from a 5K-example sweep transfer to a 60K run, and should the target be held-out loss, accuracy, reward margin, or downstream policy performance? Nathan's thread answer was still "honestly not sure." [PR #337 discussion](https://github.com/natolambert/rlhf-book/pull/337#issuecomment-4196351848)
- **Why do chat templates matter for preference reward models?** What breaks when chosen/rejected conversations are flattened, and how should train/eval template mismatch be handled? [PR #450 discussion](https://github.com/natolambert/rlhf-book/pull/450#issuecomment-4700429512)
- **How should RL algorithm choice depend on reward-model noise?** Can GRPO-style variance reduction make a bad/noisy reward look less obviously broken while leaving the real bottleneck untouched? [LinkedIn discussion](https://www.linkedin.com/posts/natolambert_reinforce-rloo-ppo-grpo-gspo-cispo-activity-7439685527253168129-kbVq)
- **How should clinical models combine verifiable rewards with human feedback where judgment remains underdetermined?** This was the most-liked recent technical YouTube comment (3 likes), though it is a theme to sharpen rather than a literal question. [Lecture 11](https://www.youtube.com/watch?v=GMry2DzC304&lc=UgyrtlBEI1Eb-OUVbjR4AaABAg)

### Hold pending evidence

- **Why did the supposedly stable/nonnegative `k3` KL estimator spike and underperform `k1`/`k2` in GRPO?** [Issue #489](https://github.com/natolambert/rlhf-book/issues/489) is a great reader-generated experiment, but the proposed three-seed rerun at `beta=0.01` has not landed. I would wait, then turn it into a results-backed correction/explanation rather than speculate in this deck.

### Separate correction worth checking

A [Lecture 9 commenter](https://www.youtube.com/watch?v=y04JhXpiI4s&lc=UgxJrdUrTAedW2vBchB4AaABAg) flags the statement around 6:05 that overfitting increases bias and decreases variance; Nathan acknowledged it. That looks worth verifying/fixing independently of Q&A 3.

Access caveat: this covers public/indexed engagement only. Deleted, held-for-review, private Discord/DM content, private analytics, and reliably attributable X reply text were not available.

File: book/templates/course.html [hunk 1]
```diff
@@ -592,7 +665,7 @@ <h1>Course</h1>
     A full course accompanying the book with added resources and other lectures I've given.
   </p>
 
-  <p class="meta">The slide decks are usually built with Claude Opus, with substantial human revisions. (Exceptions are noted on the deck itself.)</p>
+  <p class="meta">The slide decks are usually built with Claude Opus,[Colloquium](https://github.com/natolambert/colloquium) and substantial human revisions.</p>
```
[the remark this answers is not in the document]
Comment (natolambert) [reply book/templates/course.html]: Fixed in e313142 — converted to an HTML anchor with the comma spacing restored.
